DEPRECATION.md

```diff
@@ -0,0 +1,91 @@
+# Deprecation Policy Proposal
+
+As AllenNLP matures we will inevitably need to deprecate old models, interfaces and features. This document provides guidelines that attempt to minimize churn. Reducing churn helps users build next generation models with tools they've already mastered.
+
+### Terminology note
+
+We _deprecate_ some component to communicate that a preferred alternative exists. A _deprecated_ component does not necessarily have to be _removed_ from the library.
+
+### Goals
+
+1. Promote backward compatibility where not onerous.
```

Comment:
Changed to simply "Promote backward compatibility". @schmmd , let me know if that usage strikes you as being alright. I'm a bit worried saying "when reasonable" softens things too much.

Reply:
Love it.